### PR TITLE
Decrease position swap utils

### PR DESCRIPTION
**Pull request by sparqet**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature
- Testing

## What is the new behavior?

- implemented decrease_position_swap_utils.cairo file

## Does this introduce a breaking change?

No

### CODE REVIEW

File: src/position/decrease_position_swap_utils.cairo [hunk 1]
```diff
@@ -4,35 +4,58 @@
...
 use result::ResultTrait;
+use array::ArrayTrait;
 
 // Local imports.
 use satoru::position::position_utils::{
     DecreasePositionCollateralValues, UpdatePositionParams, DecreasePositionCollateralValuesOutput
 };
+use satoru::order::order::DecreasePositionSwapType;
+use satoru::position::error::PositionError;
+use satoru::swap::swap_handler::{ISwapHandlerDispatcher, ISwapHandlerDispatcherTrait};
```
Comment (Sk8erboi84) [inline src/position/decrease_position_swap_utils.cairo:15]: merge this import with the other one please

File: src/tests/position/test_decrease_position_swap_utils.cairo [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+use satoru::event::event_emitter::{IEventEmitterDispatcher, IEventEmitterDispatcherTrait};
+use satoru::data::data_store::{IDataStoreDispatcher, IDataStoreDispatcherTrait};
+use satoru::oracle::oracle::{IOracleDispatcher, IOracleDispatcherTrait};
+use satoru::bank::bank::{IBankDispatcher, IBankDispatcherTrait};
+use satoru::role::role_store::{IRoleStoreDispatcher, IRoleStoreDispatcherTrait};
+use satoru::swap::swap_utils::SwapParams;
+use core::traits::Into;
+use satoru::role::role;
+use satoru::market::market::Market;
+use satoru::position::position::Position;
```
Comment (Sk8erboi84) [inline src/tests/position/test_decrease_position_swap_utils.cairo:12]: merge this import with the 2nd one please

File: src/position/decrease_position_swap_utils.cairo [hunk 3]
```diff
@@ -4,35 +4,58 @@
...
-        execution_price: 0,
-        remaining_collateral_amount: 0,
-        base_pnl_usd: 0,
-        uncapped_base_pnl_usd: 0,
-        size_delta_in_tokens: 0,
-        price_impact_usd: 0,
-        price_impact_diff_usd: 0,
-        output: decrease_position_collateral_values_output
+    let mut swap_path_markets = ArrayTrait::new();
+    if (values.output.output_amount > 0
```
Comment (Sk8erboi84) [inline src/position/decrease_position_swap_utils.cairo:25]: The if condition is for everything not just the append

**Comment by Sk8erboi84** [comment]
Dont forget to add : "test_decrease_position_swap_utils" in lib.cairo

**Review by zarboq — APPROVED** [review]

**Pull request merged**